app/controllers/api/imports.php

```diff
@@ -0,0 +1,403 @@
+    ->inject('events')
+    ->action(function (array $resources, string $endpoint, string $apiKey, string $databaseHost, string $username, string $password, int $port, Response $response, Database $dbForProject, Document $project, Document $user, Event $eventsInstance) {
+        $import = $dbForProject->createDocument('imports', new Document([
+            '$id' => ID::unique(),
+            'status' => 'pending',
+            'stage' => 'init',
+            'source' => json_encode([
+                'type' => 'supabase',
+                'endpoint' => $endpoint,
+                'apiKey' => $apiKey,
+                'databaseHost' => $databaseHost,
+                'username' => $username,
+                'password' => $password,
+                'port' => $port,
+            ]),
+            'resources' => $resources,
+            'statusCounters' => '{}',
+            'resourceData' => "{}",
+            'errorData' => ""
+        ]));
+
+        $eventsInstance->setParam('importId', $import->getId());
+
+        // Trigger Transfer
+        $event = new Import();
+        $event
+            ->setImport($import)
+            ->setProject($project)
+            ->setUser($user)
+            ->trigger();
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($import, Response::MODEL_IMPORT);
+    });
+
+App::post('/v1/imports/nhost')
+    ->groups(['api', 'imports'])
+    ->desc('Import NHost Data')
+    ->label('scope', 'imports.write')
+    ->label('event', 'imports.create')
+    ->label('audits.event', 'import.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'imports')
+    ->label('sdk.method', 'importNhost')
+    ->label('sdk.description', '/docs/references/imports/import-nhost.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_IMPORT)
+    ->param('resources', [], new ArrayList(new WhiteList(Transfer::ALL_PUBLIC_RESOURCES)), 'List of resources to import')
+    ->param('subdomain', '', new URL(), "Source's Subdomain")
+    ->param('region', '', new Text(512), "Source's Region")
+    ->param('adminSecret', '', new Text(512), "Source's Admin Secret")
+    ->param('database', '', new Text(512), "Source's Database Name")
+    ->param('username', '', new Text(512), "Source's Database Username")
+    ->param('password', '', new Text(512), "Source's Database Password")
+    ->param('port', 5432, new Integer(), "Source's Database Port", true)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->inject('events')
+    ->action(function (array $resources, string $subdomain, string $region, string $adminSecret, string $database, string $username, string $password, int $port, Response $response, Database $dbForProject, Document $project, Document $user, Event $eventsInstance) {
```

Comment:
```suggestion
    ->action(function (array $resources, string $subdomain, string $region, string $adminSecret, string $database, string $username, string $password, int $port, Response $response, Database $dbForProject, Document $project, Document $user, Event $events) {
```

Should this work? why did we go with a different name?

Reply:
Mainly just to simplify the variable name to just one word, we can roll it back if you think it's too vague?